diff --git a/boards/nucleo-l073rz/include/periph_conf.h b/boards/nucleo-l073rz/include/periph_conf.h
index ebb2b0dc3..5ae8043fe 100644
--- a/boards/nucleo-l073rz/include/periph_conf.h
+++ b/boards/nucleo-l073rz/include/periph_conf.h
@@ -152,16 +152,22 @@ static const spi_conf_t spi_config[] = {
  * @name    ADC configuration
  * @{
  */
-#define ADC_CONFIG {            \
-    { GPIO_PIN(PORT_A, 0), 0 }, \
-    { GPIO_PIN(PORT_A, 1), 1 }, \
-    { GPIO_PIN(PORT_A, 4), 4 }, \
-    { GPIO_PIN(PORT_B, 0), 8 }, \
-    { GPIO_PIN(PORT_C, 1), 11 },\
-    { GPIO_PIN(PORT_C, 0), 10 } \
+#define ADC_CONFIG {             \
+    { GPIO_PIN(PORT_A, 0), 0 },  \
+    { GPIO_PIN(PORT_A, 1), 1 },  \
+    { GPIO_PIN(PORT_A, 4), 4 },  \
+    { GPIO_PIN(PORT_B, 0), 8 },  \
+    { GPIO_PIN(PORT_C, 1), 11 }, \
+    { GPIO_PIN(PORT_C, 0), 10 }, \
+    /* ADC VLCD channel */       \
+    { GPIO_UNDEF, 16 },          \
+    /* ADC VREF channel */       \
+    { GPIO_UNDEF, 17 },          \
+    /* ADC Temperature channel */\
+    { GPIO_UNDEF, 18 }           \
 }
 
-#define ADC_NUMOF           (6U)
+#define ADC_NUMOF           (9U)
 /** @} */
 
 /**
